File: frontend/rediscommerce-web/src/features/orders/models/order.ts
export interface OrderItem {
  productId: number
  productName: string
  unitPrice: number
  quantity: number
}

export interface Order {
  id: number
  userId: number
  status: string
  totalAmount: number
  items: OrderItem[]
  createdDate: string
}

export interface CheckoutRequest {
  userId: number
}

File: backend/RedisCommerce.Domain/Entities/Order.cs
namespace RedisCommerce.Domain.Entities;

public class Order
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public OrderStatus Status { get; set; }
    public decimal TotalAmount { get; set; }
    public DateTime CreatedDate { get; set; }
    public DateTime UpdatedDate { get; set; }

    public List<OrderItem> Items { get; set; } = [];
}
